firebaseai/src/Internal/InternalHelpers.cs

```diff
@@ -234,6 +234,20 @@ public static ModelContent ConvertToModel(this ModelContent content) {
   public static ModelContent ConvertToSystem(this ModelContent content) {
     return content.ConvertRole("system");
   }
+  
+  public static void AddIfHasValue<T>(this JsonDict jsonDict, string key,
+      T? value) where T : struct {
+    if (value.HasValue) {
+      jsonDict.Add(key, value.Value);
+    }
+  }
+  
+  public static void AddIfHasValue<T>(this JsonDict jsonDict, string key,
+      T value) where T : class {
+    if (value != null) {
+      jsonDict.Add(key, value);
+    }
+  }
```

Comment:
why we would need to functions, the only difference I can find is the ? after T in parameters

Reply:
By default, struct's are not-nullable, which is why I need the version with T?, and retrieve the underlying value with .Value.  Classes can be null, and by default doing T? with a class will result in a warning. So doing two methods helps prevent that warning.